pkg/asset/installconfig/gcp/validation.go

```diff
@@ -0,0 +1,94 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	compute "google.golang.org/api/compute/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/installer/pkg/types"
+)
+
+type validationData struct {
+	client    API
+	ic        *types.InstallConfig
+	fieldPath *field.Path
+}
+
+// Validate executes platform-specific validation.
+func Validate(client API, ic *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+	vd := &validationData{
+		client:    client,
+		ic:        ic,
+		fieldPath: field.NewPath("platform").Child("gcp"),
+	}
+	allErrs = append(allErrs, validateVPC(vd)...)
+
+	return allErrs
+}
+
+// validateVPC checks that the user-provided VPC is in the project and the provided subnets are valid.
+func validateVPC(vd *validationData) field.ErrorList {
```

Comment:
Also it seems like this function should be called `validatePlatform` as it does validation for VPC + subnets.

Reply:
I agree we should change the name from VPC, but I imagined the top level `validate` function to be the equivalent of `validatePlatform`. If there were further platform-specific data that needed to be added they would not be added here. Perhaps `validateNetwork` or `validateNetworks`?